### PR TITLE
Couple of enhancements and changes

### PR DESCRIPTION
**Pull request by Pl4gue**
- **Replace ViewFiles TextView with much more flexible `RecyclerView`**
- Replace items with `CardViews`
- Add `ScreenUtils` for **converting DIP to pixels** depending on screen size
- Add `ViewFilesDividerItemDecoration` for decorating RecyclerView items
- Change `FilesAdapter` to extend `RecyclerView.Adapter<RecyclerView.ViewHolder>`
- Change `ViewHolder` to `ViewFilesHolder` for RecyclerView Adapter, declaring elements via Butterknife
- **Structured and Cleaned Classes and Code**
- Add `Delete All Files` option in file menu  _(will be changed to action bar option according to #47 )_

### CODE REVIEW

**Comment by Pl4gue** [comment]
@Swati4star This time, I even managed to squash the 4 commits without new conflicts! 😄 😆

**Comment by Swati4star** [comment]
The UI doesn't look good in view files section
device-2017-10-30-205257
The curved edges and a lot of space in between.

Regarding Delete All feature:
Add a confirmation dialog also. In case of deleting a single also, add a confirmation dialog.
You will be adding deleteAll feature in ActionBar in some other PR?

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 1]
```diff
@@ -202,6 +236,29 @@ private void deleteFile(String name, int position) {
 
     }
 
+    private void deleteAllFiles() {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Add a comment on top of this function too.

**Comment by Pl4gue** [comment]
@Swati4star What kind of item decoration would you prefer? Just a simple dividing line?

**Comment by Pl4gue** [comment]
@Swati4star Regarding the deleting, I thought about swipe-to-delete as this would look much nicer and is relatively easy to implement with RecyclerView (one big advantage of RecyclerView  compared to previous ListView)

**Comment by Swati4star** [comment]
Yes. Just a simple divider line. Almost like the previous one.

The swipe to delete will be for deleting one file. Right?
For multiple, you will add something like a selection button to action bar. And allow user to select them and delete. Or you have something else in mind?

**Comment by Pl4gue** [comment]
Alright!
Another possible scenario would be "Files Actions" on long press and selection (and then the delete button will show up in action bar) on short press

**Comment by Swati4star** [comment]
User would prefer 'View File' to be easily available, as that must be the most used. That feature should not be on long press.

**Comment by Pl4gue** [comment]
Just changed the ItemDecoration as you wanted.

Alright, so then swipe-to-delete for single and (action bar invoked) selection delete for multiple deletions. 
I will work on the deletion as soon as you merge this.

**Comment by Swati4star** [comment]
So, this commit is a fix for https://github.com/Swati4star/Images-to-PDF/issues/44

**Comment by Pl4gue** [comment]
No, not directly. This is just an enhancement

**Comment by Swati4star** [comment]
Butterknife along with other enhancements.

**Comment by Swati4star** [comment]
device-2017-10-31-112446
UI still needs to be improved. The last divider line is different. Not a clean UI

**Comment by Pl4gue** [comment]
Okay, will try to fix this... Any other adjustments?

**Comment by Swati4star** [comment]
No. just this enhancement.

**Comment by Swati4star** [comment]
You need to add similar changes for horizontal orientation too.
I just merged another PR.

**Comment by Swati4star** [comment]
Any updates on this issue? are you still working on this one?

**Comment by Pl4gue** [comment]
Hey @Swati4star !
I actually stopped working on it, but as you just notified me, I came back to it and applied one last change and made the list styling as you wanted: Only a simple divider line.
I hope that you can merge it now! :+1: :smile:

**Comment by Pl4gue** [comment]
I would work on swipe-to-delete as well if I find enough time

**Comment by Pl4gue** [comment]
? @Swati4star :smile:

**Comment by Swati4star** [comment]
hey sorry. I didn't look at this PR. I will surely review it within this week.

**Comment by Pl4gue** [comment]
Did you review it? @Swati4star

**Comment by Pl4gue** [comment]
Aren't you going to review/merge this any time soon?

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -1,20 +1,20 @@
 <?xml version="1.0" encoding="utf-8"?>
 <android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/drawer_layout"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:fitsSystemWindows="true"
-    tools:openDrawer="start">
+                                        xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (Swati4star) [inline app/src/main/res/layout/activity_main.xml]: why this indentation change?

**Comment by Swati4star** [comment]
The PR looks great. I will merge it 👍 
Thanks for such good work.

If you wish to work on swipe to delete, you can do that in another PR 👍

**Pull request merged**